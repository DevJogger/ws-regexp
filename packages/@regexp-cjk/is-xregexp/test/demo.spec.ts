import XRegExp from 'xregexp';
import isXRegExp, { internalXRegExpData } from '../index';

test(`isXRegExp`, () =>
{

	let actual = XRegExp('a.b');

	expect(isXRegExp(actual)).toBeTruthy();
	expect(actual).toBeInstanceOf(RegExp);
	expect(actual).toMatchSnapshot();

});

test(`internalXRegExpData`, () =>
{

	let actual = XRegExp('a.b');

	expect(internalXRegExpData(actual)).toHaveProperty('source', 'a.b');
	expect(internalXRegExpData(actual)).toMatchSnapshot();

});
