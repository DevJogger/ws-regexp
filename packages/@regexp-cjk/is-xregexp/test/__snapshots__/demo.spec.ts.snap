// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`internalXRegExpData 1`] = `
Object {
  "captureNames": null,
  "flags": "",
  "source": "a.b",
}
`;

exports[`isXRegExp 1`] = `/a\\.b/`;
